logging: consolidate logging functions to always take a context

RaduBerinde opened the pull request:
Mostly mechanical change to require a context on all logging functions.

I will file issues against various components for auditing the call sites to
replace `context.TODO()` with a real context.

Part of #1779.

 @tschottdorf @petermattis The only thing to look at is `util/log/log.go`.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/7990)

<!-- Reviewable:end -->

tschottdorf commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 135 of 135 files at r1.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7990#-:-KNPExbePvp4eOFAF9gP:1417088721)*
<!-- Sent from Reviewable.io -->

RaduBerinde committed:
logging: consolidate logging functions to always take a context

Mechanical change to require a context on all logging functions.

I will file issues against various components for auditing the call sites to
replace context.TODO() with a real context.

Part of #1779.
